prow/cmd/hook/main.go

```diff
@@ -220,5 +220,20 @@ func main() {
 	// Serve plugin help information from /plugin-help.
 	http.Handle("/plugin-help", pluginhelp.NewHelpAgent(pluginAgent, githubClient))
 
-	logrus.Fatal(http.ListenAndServe(":"+strconv.Itoa(o.port), nil))
+	httpServer := &http.Server{Addr: ":" + strconv.Itoa(o.port), Handler: nil}
+
+	// Shutdown gracefully on SIGTERM or SIGINT
+	go func() {
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+		select {
+		case <-sig:
```

Comment:
The select statement is not necessary because there is only one case. You can instead just read from sig (`<-sig`) which won't return until a signal is received.